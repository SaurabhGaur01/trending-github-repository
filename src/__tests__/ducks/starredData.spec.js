import {
    setStarredData,
    clearStarredData,
    starredDataReducer,
} from '../../ducks/starredData';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('starredData Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setStarredData should match the snapshot' , () => {
            expect(setStarredData(mockObject)).toMatchSnapshot();
        });
        it('clearStarredData should match the snapshot' , () => {
            expect(clearStarredData()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(starredDataReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(starredDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(starredDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value of the API call' , () => {
            expect(starredDataReducer('', setStarredData(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearStarredData' , () => {
            expect(starredDataReducer(mockObject, clearStarredData())).toEqual([]);
        });
    });
});