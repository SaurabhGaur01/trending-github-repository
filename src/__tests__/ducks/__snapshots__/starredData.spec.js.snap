// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starredData Duck Tests Action Generator clearStarredData should match the snapshot 1`] = `
Object {
  "type": "modules/repo-app/CLEAR_STARRED_DATA",
}
`;

exports[`starredData Duck Tests Action Generator setStarredData should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/repo-app/SET_STARRED_DATA",
}
`;
